common/djangoapps/track/views/__init__.py

```diff
@@ -156,22 +143,14 @@ def task_track(request_info, task_info, event_type, event, page=None):
 
     # supplement event information with additional information
     # about the task in which it is running.
-    full_event = dict(event, **task_info)
-
-    # Get values from the task-level
-    # information, or just add placeholder values.
-    with eventtracker.get_tracker().context('edx.course.task', contexts.course_context_from_url(page)):
-        event = {
-            "username": request_info.get('username', 'unknown'),
-            "ip": request_info.get('ip', 'unknown'),
-            "event_source": "task",
-            "event_type": event_type,
-            "event": full_event,
-            "agent": request_info.get('agent', 'unknown'),
-            "page": page,
-            "time": datetime.datetime.utcnow().replace(tzinfo=pytz.utc),
-            "host": request_info.get('host', 'unknown'),
-            "context": eventtracker.get_tracker().resolve_context(),
-        }
-
-    log_event(event)
+    data = dict(event, **task_info)
+
+    context_override = contexts.course_context_from_url(page)
+    context_override['username'] = request_info.get('username', 'unknown')
+    context_override['event_source'] = 'task'
+    context_override['page'] = page
+
+    with eventtracker.get_tracker().context('edx.course.task', context_override):
+        eventtracker.emit(name=event_type, data=data)
+
+    return HttpResponse('success')
```

Comment:
@danialmalik 
why need to return HttpResponse?

Reply:
My bad. Left it there by mistake.